torch/csrc/jit/tensorexpr/kernel.cpp

```diff
@@ -1995,6 +2003,111 @@ Tensor* TensorExprKernel::computeSoftmax(
            result->stmt()}));
 }
 
+Tensor* TensorExprKernel::computeCatWoConditionals(const torch::jit::Value* v) {
+  auto const& n = v->node();
+  auto inputs = n->inputs()[0]->node()->inputs();
+  if (inputs.size() == 0) {
+    throw std::runtime_error("Empty input list is passed to aten::cat");
+  }
+
+  // Some of the inputs can be empty tensors, we need to skip them
+  // when we construct the expression, but we need to take them into
+  // account in dtype promotion.
+  std::vector<const torch::jit::Value*> nonempty_inputs;
+  for (auto input : inputs) {
+    if (input->type()->kind() == TypeKind::TensorType) {
+      auto tt = input->type()->cast<TensorType>();
+      if (tt->isComplete() && tt->sizes().size() && tt->sizes()[0] &&
+          *tt->sizes()[0]) {
+        nonempty_inputs.push_back(input);
+      }
+    }
+  }
+
+  // Promote input types.
+  // Note that we need to consider all inputs, including empty - they
+  // also affect the resultant dtype.
+  auto maybe_dtype = findDtypeForValue(inputs[0]);
+  TORCH_INTERNAL_ASSERT(
+      maybe_dtype, "Cannot find dtype for one of aten::cat inputs");
+  ScalarType highType = *maybe_dtype;
+  for (const auto input : inputs) {
+    auto maybe_dtype = findDtypeForValue(input);
+    TORCH_INTERNAL_ASSERT(
+        maybe_dtype, "Cannot find dtype for one of aten::cat inputs");
+    highType = promoteTypes(highType, *maybe_dtype);
+  }
+
+  // Now we build one loop per input:
+  //
+  // for i
+  //   for j
+  //     for k
+  //       output[i,j,k] = inp1[i,j,k]
+  // for i
+  //   for j
+  //     for k
+  //       output[i,j+l1,k] = inp2[i,j,k]
+  // for i
+  //   for j
+  //     for k
+  //       output[i,j+l2,k] = inp3[i,j,k]
+
+  auto output_sizes = inferSizesForValue(v);
+  std::vector<const Expr*> output_sizes_expr(output_sizes.size());
+  for (size_t i = 0; i < output_sizes.size(); ++i) {
+    output_sizes_expr[i] = output_sizes[i].node();
+  }
+  auto output_buf = new Buf(
+      new Var("aten_cat", kHandle), output_sizes_expr, ToDtype(highType));
```

Comment:
Nit: No need to create `Var` directly, you could pass a name to a `Buf`'s constructor directly:
```
auto output_buf = new Buf("aten_cat", output_sizes_expr, ToDtype(highType));
```

Reply:
Done.